applications/GeoMechanicsApplication/tests/test_integration_node_extrapolation.py

```diff
@@ -0,0 +1,92 @@
+        test_helper.run_kratos(file_path)
+        reader     = test_helper.GiDOutputFileReader()
+        simulation_output = reader.read_output_from(os.path.join(file_path, test_name+'.post.res'))
+        heads             = test_helper.GiDOutputFileReader.nodal_values_at_time("HYDRAULIC_HEAD", 1, simulation_output,
+                                                                             node_ids=[2, 4])
+        expected_heads    = [ 0., 1. ]
+        for head, expected_head in zip(heads, expected_heads):
+            self.assertAlmostEqual(head, expected_head, places=4)
+
+    def test_6_noded_triangle(self):
+        directory  = 'test_integration_node_extrapolation'
+        test_name = '6_noded_triangle'
+        file_path = test_helper.get_file_path(os.path.join(directory, test_name))
+        test_helper.run_kratos(file_path)
+        reader     = test_helper.GiDOutputFileReader()
+        simulation_output = reader.read_output_from(os.path.join(file_path, test_name+'.post.res'))
+        heads             = test_helper.GiDOutputFileReader.nodal_values_at_time("HYDRAULIC_HEAD", 1, simulation_output,
+                                                                             node_ids=[1, 3, 5])
+        expected_heads    = [ 1., 0.5, 0. ]
+        for head, expected_head in zip(heads, expected_heads):
+            self.assertAlmostEqual(head, expected_head, places=4)
+
+    def test_8_noded_quadrilateral(self):
+        directory  = 'test_integration_node_extrapolation'
+        test_name = '8_noded_quadrilateral'
+        file_path = test_helper.get_file_path(os.path.join(directory, test_name))
+        test_helper.run_kratos(file_path)
+        reader     = test_helper.GiDOutputFileReader()
+        simulation_output = reader.read_output_from(os.path.join(file_path, test_name+'.post.res'))
+        heads             = test_helper.GiDOutputFileReader.nodal_values_at_time("HYDRAULIC_HEAD", 1, simulation_output,
+                                                                             node_ids=[2, 4, 8])
+        expected_heads    = [ 1.09636, 0.292984, 1.29298]
+        for head, expected_head in zip(heads, expected_heads):
+            self.assertAlmostEqual(head, expected_head, places=4)
+
+
+        cauchy_stress_tensors             = test_helper.GiDOutputFileReader.nodal_values_at_time("CAUCHY_STRESS_TENSOR", 1, simulation_output,
+                                                                                     node_ids=[2, 4, 8])
+        expected_tensors    = [ [-1501.6, -6006.41, -1501.6, -376.767, 0, 0],
+                                [733.933, 2935.73, 733.933, -370.876, 0, 0],
+                                [-8.04944, -32.1978, -8.04944, -373.822, 0, 0]]
+        for cauchy_stress_tensor, expected_tensor in zip(cauchy_stress_tensors, expected_tensors):
+            for (cauchy_stress, expected) in zip(cauchy_stress_tensor, expected_tensor):
```

Comment:
A pair of redundant parenthesis:
```suggestion
            for cauchy_stress, expected in zip(cauchy_stress_tensor, expected_tensor):
```